msrestazure/azure_operation.py

```diff
@@ -182,122 +200,59 @@ def _get_body_status(self, response):
         """
         Attempt to get provisioning state from resource.
+        :param requests.Response response: latest REST call response.
         :returns: Status if found, else 'None'.
         """
-        try:
-            return self.resource.provisioning_state
-        except AttributeError:
-            pass
-        try:
-            return self.resource.properties.provisioning_state
-        except AttributeError:
+        if self._is_empty(response):
             return None
-
-    def _object_from_response(self, response):
-        """If deserialization fails, attempt to create object from
-        response body regardless.
-        Required functionality for Azure LRO's....
-
-        :param requests.Response response: latest REST call response.
-        """
         body = response.json()
-        body = {self._convert.sub(r'\1_\2', k).lower(): v
-                for k, v in body.items()}
-        properties = body.get('properties')
-        if properties:
-            properties = {self._convert.sub(r'\1_\2', k).lower(): v
-                          for k, v in properties.items()}
-            del body['properties']
-            body.update(properties)
-            self.resource = SimpleResource(**body)
-        else:
-            self.resource = SimpleResource(**body)
-
-    def _process_status(self, response):
-        """Process response based on specific status code.
-
-        :param requests.Response response: latest REST call response.
-        """
-        process = getattr(self, '_status_' + str(response.status_code))
-        process(response)
-
-    def _status_200(self, response):
-        """Process response with status code 200.
-
-        :param requests.Response response: latest REST call response.
-        """
-        status = self._get_body_status(response)
-        self.status = status if status else 'Succeeded'
-        if not status:
-            try:
-                # Even if this fails, status '200' should be successful.
-                self._deserialize(response)
-            except CloudError:
-                if self.method in ['PUT', 'PATCH']:
-                    self._object_from_response(response)
-
-    def _status_201(self, response):
-        """Process response with status code 201.
-
-        :param requests.Response response: latest REST call response.
-        :raises: BadResponse if response deserializes to CloudError.
-        """
-        try:
-            self._deserialize(response)
-            if not self.status:
-                self.status = 'Succeeded'
-        except CloudError as err:
-            raise BadResponse(str(err))
-
-    def _status_202(self, response):
-        """Process response with status code 202.
-        Just sets status to 'InProgress'.
-
-        :param requests.Response response: latest REST call response.
-        """
-        self.status = 'InProgress'
+        return body.get("properties", {}).get("provisioningState")
```

Comment:
Would it be worth using `dict.setdefault()` for `body.get("properties", {})`? Or is this the only time you care about that specific key existing?